itests/org.openhab.binding.systeminfo.tests/src/main/java/org/openhab/binding/systeminfo/test/SystemInfoOSGiTest.java

```diff
@@ -153,50 +157,54 @@ public void setUp() {
 
         registerService(mockedSystemInfo);
 
+        final SystemInfoHandlerFactory factory = getService(ThingHandlerFactory.class, SystemInfoHandlerFactory.class);
```

Comment:
I don’t understand why you move this outside of the assert. The whole point is that the getService call should retry if the ThingHandlerFactory is not ready yet. With this construct, you do not retry, you just fail if it is not ready.